controllers/eip_controller.go

```diff
@@ -234,12 +263,22 @@ func (r *EIPReconciler) reconcileTags(ctx context.Context, eip *awsv1alpha1.EIP,
 		}
 	}
 
+	convertToStringSlice := func(tags []*ec2.Tag) []string {
```

Comment:
Might be easier overall to work with tags as maps as long as possible and only convert to `ec2.Tag` when necessary.